srtcore/utilities.h

```diff
@@ -843,6 +843,126 @@ struct CallbackHolder
 };
 
 #define CALLBACK_CALL(holder,...) (*holder.fn)(holder.opaque, __VA_ARGS__)
+// The version of std::tie from C++11, but for pairs only.
+template <class T1, class T2>
+struct PairProxy
+{
+    T1& v1;
+    T2& v2;
+
+    PairProxy(T1& c1, T2& c2): v1(c1), v2(c2) {}
+
+    void operator=(const std::pair<T1, T2>& p)
+    {
+        v1 = p.first;
+        v2 = p.second;
+    }
+};
+
+template <class T1, class T2> inline
+PairProxy<T1, T2> Tie2(T1& v1, T2& v2)
+{
+    return PairProxy<T1, T2>(v1, v2);
+}
```

Comment:
👍

Reply:
Yeah, nice thing, although C++11 has this already and C++17 provides even a better alternative.